### PR TITLE
Convert authentication code to Kotlin

### PR DESCRIPTION
**Pull request by McSinyx**
Lord concurrency in Java sucks.

### CODE REVIEW

**Comment by 9a24f0** [comment]
-1 from me since everything is in Java and one thing is in Kotlin )-:

File: app/src/main/java/vn/edu/usth/minigh/AuthActivity.java [hunk 1]
```diff
@@ -1,44 +0,0 @@
...
-//    if view has focus, hiden the keyboard
-    @Override
-    public boolean dispatchTouchEvent(MotionEvent ev) {
-        if (getCurrentFocus() != null) {
-            InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
-            imm.hideSoftInputFromWindow(getCurrentFocus().getWindowToken(), 0);
-        }
-        return super.dispatchTouchEvent(ev);
-    }
-}
```
Comment (9a24f0) [inline app/src/main/java/vn/edu/usth/minigh/AuthActivity.java:44]: I don't even rmb writing this :(

File: app/src/main/java/vn/edu/usth/minigh/SplashActivity.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            delay(2000)
+            val manager = SessionManager(getApplicationContext())
+            val intent = if (manager.getSession() == -1) {
+                Intent(this@SplashActivity, AuthActivity::class.java)
+            } else {
+                Intent(this@SplashActivity, ProfileActivity::class.java)
+            }
+            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+            startActivity(intent)
+        }
```
Comment (McSinyx) [inline app/src/main/java/vn/edu/usth/minigh/SplashActivity.kt:26]: In offline discussion with @Huy-Ngo, we might want to convert to Kotlin eventually for anything that does networking for the ease of concurrency.  Notice how the snippet above can have sleep (delay) and UI modification in the same block without any sort of callback: it's a lot cleaner and easier to read in the long run.

**Comment by Huy-Ngo** [comment]
As it didn't break anything, I'm fine with this.

**Review by Huy-Ngo — APPROVED** [review]

**Pull request merged**